Source/themes.cpp

```diff
@@ -609,56 +593,59 @@ void Theme_MonstPit(int t)
  */
 void Theme_SkelRoom(int t)
 {
-	int xp, yp, i;
 	char monstrnd[4] = { 6, 7, 3, 9 };
 
 	TFit_SkelRoom(t);
 
-	xp = themex;
-	yp = themey;
+	int xp = themex;
+	int yp = themey;
 
 	AddObject(OBJ_SKFIRE, xp, yp);
 
 	if (GenerateRnd(monstrnd[leveltype - 1]) != 0) {
-		i = PreSpawnSkeleton();
+		int i = PreSpawnSkeleton();
 		SpawnSkeleton(i, { xp - 1, yp - 1 });
 	} else {
 		AddObject(OBJ_BANNERL, xp - 1, yp - 1);
 	}
 
-	i = PreSpawnSkeleton();
-	SpawnSkeleton(i, { xp, yp - 1 });
+	{

```

Comment:
Why the anonymous scope?

Reply:
To allow a consistent style for `int i = PreSpawnSkeleton();` and not reuse the same variable in a different context (bug prone).

Reply:
I think it is a poor choice. Anonymous scopes are fairly alien things as far as I'm concerned.

Would propose we refactor this into something that avoids the need for the scopes instead (in another change perhaps).

Reply:
yeah, the options i felt where resonable where:
- call the one on 643 `j`
- use anon scope
- inline all calls to `PreSpawnSkeleton()`
- reuse `i`

All of them have down sides, in the end i opted for the one that looked the most consistent with the least diff.